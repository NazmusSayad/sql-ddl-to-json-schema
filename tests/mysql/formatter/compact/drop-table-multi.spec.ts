import { getCompactFormat } from '../../parse-handler';
import { run } from '../../runner';
import createTable from './sql/create-table';

const sql = [createTable, 'DROP TABLE house, pet;'];

run(getCompactFormat, {
  'Compact formatter: Should drop two tables.': {
    queries: [sql.join('')],
  },
});
