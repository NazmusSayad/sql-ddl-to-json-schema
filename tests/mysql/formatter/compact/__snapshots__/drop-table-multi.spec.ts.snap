// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compact formatter: Should drop two tables. 1`] = `
[
  {
    "name": "person",
    "columns": [
      {
        "name": "id",
        "type": {
          "datatype": "int",
          "width": 4
        },
        "options": {
          "unsigned": true,
          "zerofill": true,
          "nullable": false,
          "autoincrement": true,
          "comment": "primary key test"
        }
      },
      {
        "name": "name",
        "type": {
          "datatype": "text",
          "length": 65535
        },
        "options": {
          "charset": "utf8",
          "collation": "utf8_general_ci",
          "nullable": false,
          "default": "Jane \\"Doe\\" O\`neil"
        }
      },
      {
        "name": "nickname",
        "type": {
          "datatype": "varchar",
          "length": 20
        },
        "options": {
          "nullable": true,
          "default": "J"
        }
      },
      {
        "name": "binarynickname",
        "type": {
          "datatype": "varchar",
          "length": 20,
          "binaryCollation": true
        },
        "options": {
          "nullable": false
        }
      },
      {
        "name": "ssn",
        "type": {
          "datatype": "int",
          "width": 2
        },
        "options": {
          "nullable": true,
          "invisibleWithSystemVersioning": true
        }
      },
      {
        "name": "ssn1",
        "type": {
          "datatype": "int",
          "width": 2
        },
        "options": {
          "unsigned": true,
          "nullable": true,
          "comment": "SMALLINT UNSIGNED"
        }
      },
      {
        "name": "ssn2",
        "type": {
          "datatype": "int",
          "width": 1
        },
        "options": {
          "nullable": true,
          "comment": "TINYINT"
        }
      },
      {
        "name": "ssn3",
        "type": {
          "datatype": "int",
          "width": 1
        },
        "options": {
          "unsigned": true,
          "nullable": true,
          "comment": "TINYINT UNSIGNED"
        }
      },
      {
        "name": "ssn4",
        "type": {
          "datatype": "int",
          "width": 4
        },
        "options": {
          "nullable": true,
          "comment": "INT"
        }
      },
      {
        "name": "ssn5",
        "type": {
          "datatype": "int",
          "width": 4
        },
        "options": {
          "unsigned": true,
          "nullable": true,
          "comment": "INT UNSIGNED"
        }
      },
      {
        "name": "ssn6",
        "type": {
          "datatype": "int",
          "width": 3
        },
        "options": {
          "nullable": true,
          "comment": "MEDIUMINT"
        }
      },
      {
        "name": "ssn7",
        "type": {
          "datatype": "int",
          "width": 3
        },
        "options": {
          "unsigned": true,
          "nullable": true,
          "comment": "MEDIUMINT UNSIGNED"
        }
      },
      {
        "name": "ssn8",
        "type": {
          "datatype": "int",
          "width": 8
        },
        "options": {
          "nullable": true,
          "comment": "BIGINT"
        }
      },
      {
        "name": "ssn9",
        "type": {
          "datatype": "int",
          "width": 8
        },
        "options": {
          "unsigned": true,
          "nullable": true,
          "comment": "BIGINT UNSIGNED"
        }
      },
      {
        "name": "height",
        "type": {
          "datatype": "decimal",
          "digits": 10,
          "decimals": 0
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "weight",
        "type": {
          "datatype": "decimal",
          "digits": 3,
          "decimals": 0
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "status",
        "type": {
          "datatype": "bit",
          "length": 1
        },
        "options": {
          "nullable": true,
          "invisibleWithoutSystemVersioning": true
        }
      },
      {
        "name": "dob",
        "type": {
          "datatype": "date",
          "fractional": 0
        },
        "options": {
          "nullable": true,
          "invisible": true
        }
      },
      {
        "name": "tob",
        "type": {
          "datatype": "time",
          "fractional": 3
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "initials",
        "type": {
          "datatype": "char",
          "length": 3
        },
        "options": {
          "nullable": true,
          "format": "fixed"
        }
      },
      {
        "name": "prefix",
        "type": {
          "datatype": "char",
          "length": 3
        },
        "options": {
          "nullable": true,
          "format": "fixed"
        }
      },
      {
        "name": "suffix",
        "type": {
          "datatype": "char",
          "length": 1
        },
        "options": {
          "nullable": true,
          "format": "fixed"
        }
      },
      {
        "name": "nsuffix",
        "type": {
          "datatype": "varchar",
          "length": 3
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "sequence",
        "type": {
          "datatype": "varbinary",
          "length": 20
        },
        "options": {
          "nullable": true,
          "storage": "disk"
        }
      },
      {
        "name": "avatar",
        "type": {
          "datatype": "blob",
          "length": 65535
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "motto",
        "type": {
          "datatype": "text",
          "length": 50
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "history",
        "type": {
          "datatype": "text",
          "length": 255
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "gender",
        "type": {
          "datatype": "enum",
          "values": [
            "M",
            "F"
          ]
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "shape",
        "type": {
          "datatype": "multipoint"
        },
        "options": {
          "nullable": true
        }
      },
      {
        "name": "pattern",
        "type": {
          "datatype": "multilinestring"
        },
        "options": {
          "nullable": true
        }
      }
    ],
    "primaryKey": {
      "columns": [
        {
          "column": "id"
        }
      ]
    },
    "uniqueKeys": [
      {
        "columns": [
          {
            "column": "ssn"
          }
        ]
      },
      {
        "columns": [
          {
            "column": "motto",
            "length": 50
          }
        ],
        "name": "u_motto",
        "indexType": "btree",
        "options": {
          "comment": "be original"
        }
      }
    ],
    "fulltextIndexes": [
      {
        "columns": [
          {
            "column": "initials"
          }
        ],
        "name": "fi_initials"
      }
    ],
    "options": {
      "autoincrement": 2,
      "avgRowLength": 250,
      "charset": "utf8",
      "checksum": 1,
      "collation": "utf8_general_ci",
      "comment": "test table",
      "compression": "zlib",
      "engine": "XtraDB",
      "insertMethod": "last",
      "union": [
        "pet"
      ]
    }
  }
]
`;
